src/QY2ListView.cc

```diff
@@ -418,29 +412,55 @@ QY2ListViewItem::operator< ( const QTreeWidgetItem & otherListViewItem ) const
 	{
 	    return ( this->serial() < otherCheckListItem->serial() );
 	}
-
     }
 
-    // numeric sorting if columns are numbers
     int column = treeWidget()->sortColumn();
-    QString text1=text(column).trimmed();
-    QString text2=otherListViewItem.text(column).trimmed();
 
-    text1=text1.left(text1.indexOf(QChar(' ')));
-    text2=text2.left(text2.indexOf(QChar(' ')));
+    if (other)
+    {
+        return compare(smartSortKey(column), other->smartSortKey(column));
+    }
+
+    return compare(text(column).trimmed(), otherListViewItem.text(column).trimmed());
+}
+
+
+bool
+QY2ListViewItem::sortByInsertionSequence() const
+{
+    QY2ListView * parentListView = dynamic_cast<QY2ListView *> (treeWidget());
+
+    if ( parentListView )
+	return parentListView->sortByInsertionSequence();
+
+    return false;
+}
+
+
+bool
+QY2ListViewItem::compare(const QString& text1, const QString& text2) const
+{
+    // numeric sorting if columns are numbers
 
     bool ok1, ok2; // conversion to int successful
-    bool retval = text1.toInt(&ok1) < text2.toInt(&ok2);
+    bool retval = text1.toLongLong(&ok1) < text2.toLongLong(&ok2);
 
-    if (ok1 && ok2 )
+    if (ok1 && ok2)
         return retval;     // int < int
     else if (ok1 && !ok2)
         return true;       // int < string
     else if (!ok1 && ok2)
         return false;      // string > int
 
-    // and finally non-numeric sorting is done by the base class
-    return QTreeWidgetItem::operator<(otherListViewItem);
+    // and finally non-numeric sorting is done locale aware
+    return QString::localeAwareCompare(text1, text2) < 0;   // string < string
```

Comment:
just curious is ncurses sorting same regarding locales? ( I mean also locale aware and if it is use same algorithm in background? )

Reply:
ncurses use wcscoll which is also locale aware. What Qt does internally I do not know.